src/main/java/org/opensearch/ad/ml/CheckpointDao.java

```diff
@@ -483,6 +553,11 @@ private void logFailure(BulkByScrollResponse response, String detectorID) {
                     // avoid possible null pointer exception
                     samples = new ArrayDeque<>();
                 }
+                ThresholdedRandomCutForest trcf = null;
+                if (json.has(ENTITY_TRCF)) {
+                    trcf = toTrcf(json.getAsJsonPrimitive(ENTITY_TRCF).getAsString());
+                }
+                // TODO: convert rcf and threshold to trcf
```

Comment:
Could you put line 560~571 under the else branch? Want to make it explicit that we only expect trcf or rcf/threshold, but not both.

Reply:
yes, updated it.